scripts/jest/ts-preprocessor.js

```diff
@@ -71,18 +71,17 @@ function compile(content, contentFilename) {
     getNewLine: function() {
       return '\n';
     },
+    fileExists: function (filename) {
+      return ts.sys.fileExists(filename);
+    },
   };
   var program = ts.createProgram([
     'lib.d.ts',
     'jest.d.ts',
     contentFilename,
   ], tsOptions, compilerHost);
-  var errors = program.getDiagnostics();
-  if (!errors.length) {
-    var checker = program.getTypeChecker(true);
-    errors = checker.getDiagnostics();
-    checker.emitFiles();
-  }
+  var emitResult = program.emit();
+  var errors = ts.getPreEmitDiagnostics(program).concat(emitResult.diagnostics);
```

Comment:
... which in an alternative universe might have prevented the spreadgate. :smile: